crates/forge/tests/integration/gas.rs

```diff
@@ -697,11 +701,17 @@ fn l1_handler_cost() {
 
     let result = run_test_case(&test);
     assert_passed(&result);
-    // FIXME these values changed
+    // TODO(#2960)
     // 96 = gas cost of onchain data (deploy cost)
     // int(5.12 * 4) = 21 = keccak cost from l1 handler
-    // 14643 - l1 cost of payload + emit message handle event
-    assert_gas(&result, "l1_handler_cost", 96 + 15944);
+    // in this test, l1_handler_payload_size = 6
+    // 15923 = 12251 (gas used for processing L1<>L2 messages on L1) + 3672 (SHARP gas, 6 * 612)
+    // 12251 = 3072 (6 * 512, 512 is gas per memory word) +
+    //         + 4179 (result of get_consumed_message_to_l2_emissions_cost(6) which is get_event_emission_cost(3, 3 + 6) = 375 + (3 + 1) * 375 + 9 * 256) +
+    //         + 0 +
+    //         + 5000 (1 * 5000, 5000 is gas per counter decrease)
```

Comment:
`gas per counter decrease` what do you mean by that?

Reply:
It's related with l1 storage write cost - in docs it says we pay 20000 but in fact we get a refund for 15000, so it's 5000 left. I've added permalink to blockifier [code comment](https://github.com/starkware-libs/sequencer/blob/main/crates/blockifier/src/fee/resources.rs#L364-L368).